lib/granite/action/precondition.rb

```diff
@@ -0,0 +1,38 @@
+module Granite
+  class Action
+    class Precondition
```

Comment:
@dziulius Should this inherit from Ruby's `BasicObject` instead of implicit inheriting from `Object`?

We want to forward all the calls to `@context` via `#method_missing`, the less methods are defined for `Granite::Action::Precondition` the higher chance of success.

Given `BasicObject` has less methods of `Object`, I would suggest to use it as superclass.